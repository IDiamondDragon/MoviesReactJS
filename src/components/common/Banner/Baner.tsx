import React from "react";

import { useRedirectToSearchPage } from '../../../services/hooks/useRedirectToSearchPage';

import baner from "../../../assets/images/shared/Banner/netflix.png";
import styles from './Baner.module.scss';



export interface BanerProps {
  className: string
}

export function Baner({ className }: BanerProps): JSX.Element {
  const redirectToSearchPage = useRedirectToSearchPage();
  
  return (
    <img className={`${styles.baner} ${className}`} src={baner} onClick={redirectToSearchPage}/>
  );
}

export default Baner