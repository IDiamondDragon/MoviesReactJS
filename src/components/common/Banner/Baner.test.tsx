import React from 'react';
import { render } from '@testing-library/react';
import Baner from './Baner';


jest.mock('../../../services/hooks/useRedirectToSearchPage', () => {
  return {
      useRedirectToSearchPage: jest.fn().mockImplementation(() => {
      return () => ({});
    }),
  };
});


describe('Baner', () => {

  test("renders correctly", () => {
      const { asFragment } = render(<Baner className="" />);

      expect(asFragment()).toMatchSnapshot();
  })

});
