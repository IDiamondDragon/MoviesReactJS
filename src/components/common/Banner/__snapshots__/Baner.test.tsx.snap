// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Baner renders correctly 1`] = `
<DocumentFragment>
  <img
    class="baner "
    src="test-file-stub"
  />
</DocumentFragment>
`;
